File: my_01_classesJavaAndJsComparison/src/Employee.java
public class Employee {
	/*
	 * Static attributes
	 */
	public static int max = 99;

	/*
	 * Static methods
	 */
	public static void log() {
		System.out.println("Logging...");
	}

	/*
	 * Constructor
	 */
	public Employee(String fullName) {
		this.fullName = fullName;
	}

	public Employee() {
	}

	/*
	 * Attributes
	 */
	private String fullName;

	/*
	 * Methods
	 */
	public String getFullName() {
		return fullName;
	}

	public void setFullName(String fullName) {
		this.fullName = fullName;
	}

	public int generateId() {
		return this.fullName.hashCode();
	}

	public static void main(String[] args) {
		Employee.log(); // logs "Logging..."

		Employee emp1 = new Employee("John Doe");
		System.out.println("emp1=" + emp1.getFullName());

		Employee emp2 = new Employee();
		emp2.setFullName("Jane Doe");
		System.out.println("emp2=" + emp2.getFullName());

		System.out.println("max=" + Employee.max);
		System.out.println("max=" + emp2.max);

		/*
		 * Odds
		 */
		// Employee.setFullName("Employee Class"); // you can't call it, no meaning...

		// System.out.println(Employee.getFullName()); // you can't call it, no
		// meaning...
		// traces undefined

		Employee emp3 = new Employee();
		System.out.println("emp3=" + emp3.getFullName()); // null string, as expected, correct!
	}
}

File: my_01_classesJavaAndJsComparison/js/Employee.js
/**
 * Employee class defined with the help of the Immediately-Invoked Function Expression (IIFE) mechanism
 */
var Employee = (function() {
	var claxx = function(fullName) {
		this.fullName = fullName;
	}

	claxx.log = function() {
		console.log("Logging..."+ this);
	}

	claxx.prototype = {
		setFullName : function(fullName) {
			this.fullName = fullName;
		},
		getFullName : function() {
			console.log("get..."+ this);

			return this.fullName;
		}
	};

	return claxx;
})();

Employee.log(); // logs "Logging..."

var emp1 = new Employee("John Doe");
console.log("emp1=" + emp1.getFullName());

var emp2 = new Employee();
emp2.setFullName("Jane Doe");
console.log("emp2=" + emp2.getFullName());

/*
 * Odds
 */
Employee.prototype.setFullName("Employee Class"); // you can call it, no meaning... 
console.log(Employee.prototype.getFullName()); // you can call it, no meaning... traces undefined
var emp3 = new Employee();
console.log("emp3=" + emp3.getFullName());  // undefined, correct!
